sources/core/Stride.Core.DiagnosticsServices/PropertyAttributeFinder.cs

```diff
@@ -0,0 +1,48 @@
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using System.Reflection;
+using Microsoft.CodeAnalysis.CSharp;
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+
+namespace StrideDiagnostics;
+internal class PropertyAttributeFinder
+{
+    /// <summary>
+    /// Walks through a base class of a class and retrieves all allowed Properties.
+    /// Then it tries to get it's own base class and get from it all allowed Properties recursively.
+    /// All the Properties get summed up.
+    /// If they fulfill :
+    ///     Proper Accessors <see cref="PropertyHasAllowedAccessors(IPropertySymbol)"/>
+    ///     Shouldnt be Ignored <see cref="HasDataMemberIgnoreAttribute(IPropertySymbol)"/>
+    /// </summary>
+    /// <param name="currentBaseType">The base class which the ClassDeclarationSyntax has</param>
+    /// <returns>All allowed Properties in any base class in the inheritance tree</returns>
+    public static IEnumerable<IPropertySymbol> FilterBasePropertiesRecursive(ref INamedTypeSymbol currentBaseType)
+    {
+        List<IPropertySymbol> result = new List<IPropertySymbol>();
+        while (currentBaseType != null)
+        {
+            result.AddRange(currentBaseType.GetMembers().OfType<IPropertySymbol>().Where(PropertyHasAllowedAccessors));
+            currentBaseType = currentBaseType.BaseType;
+        }
+        return result;
+    }
+
+    private static bool PropertyHasAllowedAccessors(IPropertySymbol propertyInfo)
```

Comment:
Note that Stride serialization supports get-only properties for collections and classes where the contents are assigned to the instance. There is more complicated logic needed here. See `Stride.Core.DataMemberMode`

Reply:
currently get only properties work with the check

the assigning isnt covered yet, i gotta check it out

Reply:
could you give me an example when my current check would fail?